lib/marketingcloudsdk/client.rb

```diff
@@ -121,11 +124,19 @@ def initialize(params={}, debug=false)
 			self.refresh_token = params['refresh_token'] if params['refresh_token']
 
 			self.wsdl = params["defaultwsdl"] if params["defaultwsdl"]
```

Comment:
where is the code to read the OAuth2 options from parameters?

Reply:
This is the constructor for the Client class:

def initialize(**params**={}, debug=false)
@refresh_mutex = Mutex.new
self.debug = debug
**client_config = params['client']**

All the settings are further retrieved from the **client_config** variable.